graphql/api/src/main/resources/stream-registry.graphql

```diff
@@ -33,24 +33,48 @@ type Tag {
 
 ############ Specification ############
 
+input SecurityInput {
+    role: String!
+    principals: [String!]!
+}
+
+input SecurityQuery {
+    roleRegex: String!
+    principalRegex: String!
```

Comment:
On the query side, they should not be mandatory. E.g. you may be searching for an entity with a particular role so not supplying a principalRegex.

```suggestion
    roleRegex: String
    principalRegex: String
```